openedx/core/djangoapps/catalog/utils.py

```diff
@@ -101,18 +104,69 @@ def get_program_types(user=None):  # pylint: disable=redefined-builtin
         return []
 
 
-def get_programs_data(user=None):
-    """Return the list of Programs after adding the ProgramType Logo Image"""
+def _get_program_instructors(program):
+    """
+    Returns the list of instructor from cached if cache key exists otherwise
+    iterate over the courses and return all the instructors of each course run
+    """
+    catalog_integration = CatalogIntegration.current()
+    if catalog_integration.enabled:
+        cache_key = 'program.instructors.{program_id}'.format(
+            program_id=program.get('uuid')
+        )
+
+        instructor_lookup_list = []
+        instructors = cache.get(cache_key) or []
+        if instructors:
+            return instructors
+
+        module_store = modulestore()
+        for key in _get_all_course_run_keys(program):
+            descriptor = module_store.get_course(key)
```

Comment:
@hasnain-naveed @afzaledx On second (or third?) thought, I'm wondering if we should actually add the instructors to CourseOverview. The list of course runs associated with a program could grow long as time goes on. It would be much better if we could run a single database query to collect all the instructor info. We should also still cache the result of that query in this function.